namer/namer.py

```
@@ -192,10 +187,21 @@ def process_file(command: Command) -> Optional[Command]:
                 matched = search_results.get_match()
                 if matched:
                     new_metadata = matched.looked_up
+                else:
+                    # Prefer provider-set ambiguity flag
+                    ambiguous = getattr(search_results, 'ambiguous', False)
+                    # Fallback heuristic only if provider didn't set it
+                    if not ambiguous:
+                        try:
+                            all_results = [r for r in search_results.results if r]
+                            ambiguous = len(all_results) > 0
+                        except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)